osu.Game.Tests/Visual/SongSelect/TestSceneBeatmapInfoWedgeV2.cs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Testing;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Legacy;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Screens.Select;
+using osuTK;
+
+namespace osu.Game.Tests.Visual.SongSelect
+{
+    [TestFixture]
+    public partial class TestSceneBeatmapInfoWedgeV2 : OsuTestScene
+    {
+        private RulesetStore rulesets = null!;
+        private TestBeatmapInfoWedgeV2 infoWedge = null!;
+        private readonly List<IBeatmap> beatmaps = new List<IBeatmap>();
+
+        [BackgroundDependencyLoader]
+        private void load(RulesetStore rulesets)
+        {
+            this.rulesets = rulesets;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Add(infoWedge = new TestBeatmapInfoWedgeV2
+            {
+                State = { Value = Visibility.Visible },
+                Width = 0.6f,
+                RelativeSizeAxes = Axes.X,
+                Margin = new MarginPadding { Top = 20, Left = -10 }
+            });
+
+            AddSliderStep("change star difficulty", 0, 11.9, 5.55, v =>
+            {
+                foreach (var hasCurrentValue in infoWedge.Info.ChildrenOfType<IHasCurrentValue<StarDifficulty>>())
+                    hasCurrentValue.Current.Value = new StarDifficulty(v, 0);
+            });
+        }
+
+        [Test]
+        public void TestRulesetChange()
+        {
+            selectBeatmap(Beatmap.Value.Beatmap);
+
+            AddWaitStep("wait for select", 3);
+
+            foreach (var rulesetInfo in rulesets.AvailableRulesets)
+            {
+                var instance = rulesetInfo.CreateInstance();
+                var testBeatmap = createTestBeatmap(rulesetInfo);
+
+                beatmaps.Add(testBeatmap);
+
+                setRuleset(rulesetInfo);
+
+                selectBeatmap(testBeatmap);
+
+                testBeatmapLabels(instance);
+            }
+        }
+
+        [Test]
+        public void TestWedgeVisibility()
+        {
+            // Mostly just in case someone runs this test before others,
```

Comment:
Not entirely sure if this is OK to do? it alters the visuals somewhat, is a background behind the wedge preferable for showcasing the shadow

Reply:
Yes, change the background instead.

Reply:
I have no idea why this was resolved. It shouldn't be here. It affects the test PERMANENTLY (you are never recreating the drawables).